### PR TITLE
compliance - blocked country and user list

### PR DESCRIPTION
**Pull request by owocki**
##### Description

from a complilance perspective.. we do not want to do business with those in OFAC countries or on the OFAC list. this PR takes an initial stab at making sure those users cannot login to the platform

##### Refers/Fixes

email thread with product & legal 

##### Testing

tested locally

### CODE REVIEW

**Comment by octavioamu** [comment]
As we use github login and they already do this. Do we need to recheck?

**Comment by owocki** [comment]
thats a good question to ask legal on our call with them next week.

cc @frankchen07

File: app/dashboard/utils.py [hunk 1]
```diff
@@ -801,7 +801,31 @@ def get_tx_status(txid, network, created_on):
...
+            is_on_banned_countries = Country.objects.filter(name=last_login_country)
+            if is_on_banned_countries:
+                return True
+
+    # check banned entity list
+    if self.user:
+        first_name = self.user.first_name
+        last_name = self.user.last_name
+        full_name = '{first_name} {last_name}'
+        is_on_banned_user_list = Entitty.objects.filter(fullName__icontains=full_name)
```
Comment (danlipert) [inline app/dashboard/utils.py]: typo here

Comment (owocki) [reply app/dashboard/utils.py]: ce08234a1fd766145e869f55a834b040d28ece38

**Comment by owocki** [comment]
sounds like per the legal call its a good idea to have our own checking in place

**Comment by owocki** [comment]
i turned around the code review..

**Comment by danlipert** [comment]
> i turned around the code review..

@owocki is this you saying that its ready for final review? I see there is a note in the description that says it shouldn't be merged

**Comment by owocki** [comment]
@danlipert just did some final testing + submitted it (removing the hedge from the desc)

File: app/dashboard/utils.py [hunk 2]
```diff
@@ -801,7 +801,34 @@ def get_tx_status(txid, network, created_on):
...
 
 def is_blocked(handle):
-    return BlockedUser.objects.filter(handle__iexact=handle, active=True).exists()
+    # check admin block list
+    is_on_blocked_list = BlockedUser.objects.filter(handle__iexact=handle, active=True).exists()
+    if is_on_blocked_list:
+        return True
+
+    # check banned country list
+    from compliance.models import Country, Entity
```
Comment (danlipert) [inline app/dashboard/utils.py]: Can we import this once in the submodule?

File: app/compliance/management/commands/pull_compliance_list.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    countries = 'Balkans, Belarus, Burma, Cote D\'Ivoire , Cuba, Democratic Republic of Congo, Iran, Iraq, Liberia, North Korea, Sudan, Syria, Zimbabwe'.split(',')
+
+    # insert data
+    for country in countries:
+        Country.objects.create(name=country)
+
+
+def insert_entities():
+    # clear existing table
+    Entity.objects.all().delete()
```
Comment (danlipert) [inline app/compliance/management/commands/pull_compliance_list.py]: If we're going to ever do any kind of tracking / recording of compliance related service denial or anything like that we'll have to not rewrite this table over and over - a premature optimization for now but lets all make sure we are aware of this for the future.

Comment (owocki) [reply app/compliance/management/commands/pull_compliance_list.py]: got it; @alexvotofuture should we keep a record of OFAC denials?  would it be handy to have to show regulatory compliance?

**Comment by owocki** [comment]
@danlipert turned around ur feedback

File: app/compliance/management/commands/pull_compliance_list.py (hunk 3 above)
Comment (owocki) [reply app/compliance/management/commands/pull_compliance_list.py]: lets punt to a v2

File: app/compliance/management/commands/pull_compliance_list.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    # pull data
+    url = 'https://www.treasury.gov/ofac/downloads/consolidated/consolidated.xml'
```
Comment (thelostone-mc) [inline app/compliance/management/commands/pull_compliance_list.py]: who gave this this btw ?

Comment (owocki) [reply app/compliance/management/commands/pull_compliance_list.py]: i dont understand the question?

**Pull request merged**